content/blog/building-original-projects/index.md

```diff
@@ -0,0 +1,285 @@
+
+## Tackling Error Messages
+Error messages will be your constant companion in building projects, and at times can seem obtuse and confusing. However, behind their intimidating exterior lies a wealth of useful information. Let's see if we can take advantage of error messages to solve our problems.
+
+In order to work with error messages, we'll need to do two things:
+
+1. Locate the two parts of an error message that describe what the error is, and what line of your code the error originated on.
+2. Research and resolve the error.
+
+For the following example, I'll be using JavaScript errors, which have an `type`, `message` and `stack trace`. This is a very common anatomy for error messages, but your preferred language might have a different style of errors. This means learning how to read error messages in one language will transfer somewhat to other languages, but it is still highly recommended  to read the documentation on error messages for your chosen tools!
+
+### Anatomy of error messages
+
+In JavaScript's case, when runtime errors occur, an [Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) object is thrown holding the relevant information. Let's take a look at this block of JavaScript code:
+
+```javascript
+// declaring two numbers
+const a = 5;
+const b = 2;
+// a function that adds two numbers together and returns the result
+function add(a, b) {
+  return c;
+}
+// a function that 
+function print() {
+  const sum = add(a, b);
+  console.log(sum);
+}
+// calling the print function
+print();
+```
+
+Attempting to run it will result in this error message:
+
+![error](./error.png)
+
+What went wrong? If we take the message apart piece by piece, we'll understand.
+
+The first line, `Uncaught ReferenceError` refers to the `type`, in this case it is a [ReferenceError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError).
+
+The rest of the line, `c is not defined` refers to the `message`. This is the meaning of the error, and simply means that `c` does not exist as a variable in the current scope.
+
+The last three lines are the `stack trace`, a list of function calls that lead to the point where the error was thrown. In order to read this, we'll start at the last line:
+
+ ```
+ at script.js:13:1
+ ```
+
+This is saying that we began in the file `script.js`, at line 13, column 1. If we go to that line in our code, we see the print function is called like so:
+
+```javascript
+print();
+```
+
+From there we can move up the stack trace and read the next line:
+
+```
+at print (script.js:9:15)
+```
+
+If we look at line 9, column 15 in the same `script.js` file, we'll notice another function call, this time to `add`:
+
+```javascript
+const sum = add(a, b);
+```
+
+Now we can follow the stack trace up further:
+
+```at add (script.js:5:3)
+at add (script.js:5:3)
+```
```

Comment:
[Important] This breaks out code embed. To fix this:

````suggestion
```
at add (script.js:5:3)
```
````